pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/column/ColumnIndexContainer.java

```diff
@@ -41,21 +41,15 @@
 import com.linkedin.pinot.core.segment.store.ColumnIndexType;
 import com.linkedin.pinot.core.segment.store.SegmentDirectory;
 import java.io.IOException;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public abstract class ColumnIndexContainer {
-  private static final Logger LOGGER = LoggerFactory.getLogger(ColumnIndexContainer.class);
-
   public static ColumnIndexContainer init(SegmentDirectory.Reader segmentReader, ColumnMetadata metadata,
-      IndexLoadingConfigMetadata indexLoadingConfigMetadata)
+      IndexLoadingConfig indexLoadingConfig)
       throws IOException {
     String column = metadata.getColumnName();
     boolean loadInverted = false;
-    if (indexLoadingConfigMetadata != null) {
-      if (indexLoadingConfigMetadata.getLoadingInvertedIndexColumns() != null) {
-        loadInverted = indexLoadingConfigMetadata.getLoadingInvertedIndexColumns().contains(metadata.getColumnName());
-      }
+    if (indexLoadingConfig != null) {
+      loadInverted = indexLoadingConfig.getInvertedIndexColumns().contains(column);
```

Comment:
Because it can never be null. (Enforced by annotation. If we can clean up all the warnings in IDE, we can save a lot of unnecessary checks)